src/ReDate.re

```diff
@@ -264,6 +269,21 @@ let subYears = (date, years) => date->addYears(- years);
 
 let startOfYear = Internal.startOfYear;
 
+let isSameYear = (fst, snd) => fst->Internal.startOfYear->isEqual(snd->Internal.startOfYear);
+
+let isLeapYear = date => Date.(date->getFullYear->int_of_float->isLeap);
+
+let endOfYear = Internal.endOfYear;
+
+let lastMonthOfYear = date =>
+  Date.(makeWithYMD(~year=date->getFullYear +. 1., ~month=0., ~date=0., ())->startOfMonth);
```

Comment:
1. why don't we use `date->getFullYear->succ` instead of `date->getFullYear +. 1`? (https://reasonml.github.io/api/Pervasives.html)
2. I think using `startOfMonth` is redundant here, just create a date with a day `1` and then use `makeDateWithStartOfDayHours`